binary_gentoo/internal/cli/build.py

```diff
@@ -125,6 +126,22 @@ def parse_command_line(argv):
                         action='store_true',
                         help='enforce installation (default: build but do not install)')
 
+    parser.add_argument('--use',
+                        dest='use',
+                        default='',
+                        help='custom one-off use flags (default: none)')
+
+    parser.add_argument('--update',
```

Comment:
The idea of `--update` (at least to me) is to have a way to bring the gentoo/stage3 completely up to date with the latest packages and with the correct USE flags, and therefore use it together with `@system` or `@world`. 

The `--shy-rebuild`, as I understand it, is there to control whether the target atom is rebuilt or not, and is typically used with a specific atom, and not `@system` or `@world`. Also, unrelated to this specific topic: it may be interesting to have another argument to control whether the dependencies are rebuilt or not, e.g. `--shy-rebuild-dependencies`? Currently dependencies are always with `--usepkg=y`.

So they serve different purposes to me, although the user of course has the possibility to combine them (`--shy-rebuild --update @world`). I suppose if the user combines them like that, the system or world update will simply rebuild everything instead of using the precompiled packages. Which, I would say, is some flexibility that can be left to the user to decide what he wants.

I don't know the behavior of portage well enough to know what would happen if the user does `--shy-rebuild --update <atom>`, and if there are any conflicts in behavior there. That is probably what your are referring to?

Reply:
May let's just ignore my previous comment: Portage allows to combine them, maybe I'm worry-wondering too much here.

The reason I introduced `--shy-rebuild` was that I wanted to both be able to save (or speed up) unnecessary recompiles and be able to enforce a recompile without need to manually find and delete a specific previous binary package. I aligned with portage, which is why `--shy-rebuild` is not the default at the moment.

For `--shy-rebuild-dependencies` I haven't had a need yet, have you? For a case where existing binary packages are causing conflicts, I would mass-delete them with a command like…
```
gentoo-packages --pkgdir packages-17.1/ delete --metadata 'dev-libs/libffi[^ ]*:0/7='
```
… currently. Then nothing would stop pulling in `dev-libs/libffi:0/8`.

Reply:
The shy-rebuild-dependencies was something I was thinking about; not a specific use case at the moment. So maybe better to implement in the future if it would be needed. And indeed the mass deletion would do the same thing.

If you feel then this is OK, I'll let you resolve the conversation.